src/Sentry/Platforms/Cocoa/SentrySdk.cs

```diff
@@ -10,11 +10,15 @@ public static partial class SentrySdk
     private static void InitSentryCocoaSdk(SentryOptions options)
     {
         options.LogDebug("Initializing native SDK");
-        // Workaround for https://github.com/xamarin/xamarin-macios/issues/15252
-        ObjCRuntime.Runtime.MarshalManagedException += (_, args) =>
+
+        if (options.Native.EnableMarshalManagedException)
```

Comment:
Is "Enable" the right option here? Since we're setting it to `args.ExceptionMode = ObjCRuntime.MarshalManagedExceptionMode.UnwindNativeCode;`

Should we just set that when compiling for Native AOT, else leave it as-is?

Also note in .NET 10 this is likely to be fixed

Reply:
> Should we just set that when compiling for Native AOT, else leave it as-is?

That's a very good point! since this option is mainly for Native AOT anyway

> Also note in .NET 10 this is likely to be fixed

That would be more of a reason to leave it as an option, then. 


But honestly, now that you mention it, I'm more for leaving it out for Native AOT. Then if/when .NET 10 fixes it, we can put it back if need be